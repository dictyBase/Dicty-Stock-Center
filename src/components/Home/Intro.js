// @flow
import React, { Component } from "react"
import { connect } from "react-redux"
import Grid from "@material-ui/core/Grid"
import Skeleton, { SkeletonTheme } from "react-loading-skeleton"
import InlineEditor from "components/InlineEditor"
import { fetchInfoPage } from "actions/page"

type Props = {
  /** the Auth object taken from the current state */
  auth: Object,
  /** Action creator that fetches data from API */
  fetchInfoPage: Function,
  /** the Page object taken from the current state */
  page: Object,
  /** Checks if data is currently being fetched */
  isFetching: boolean,
}

const slugName = "dsc-intro"

/**
 * Intro fetches and displays the Intro page content.
 */

export class Intro extends Component<Props> {
  // set defaultprops to prevent console warnings
  static defaultProps = {
    page: {
      data: {
        attributes: {},
      },
    },
  }
  componentDidMount() {
    this.props.fetchInfoPage(slugName)
  }
  render() {
    const { isFetching, page } = this.props

    if (!isFetching && page.data.attributes.content) {
      return (
        <InlineEditor
          auth={this.props.auth}
          page={this.props.page}
          slug={slugName}
        />
      )
    }
    return (
      <Grid container justify="center">
        <Grid item xs={12}>
          <SkeletonTheme color="#D3D3D3	" highlightColor="#DCDCDC">
            <Skeleton count={5} />
            <br />
            <br />
          </SkeletonTheme>
        </Grid>
      </Grid>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  isFetching: state.page.isFetching,
  page: state.page[slugName],
})

export default connect(
  mapStateToProps,
  { fetchInfoPage },
)(Intro)
