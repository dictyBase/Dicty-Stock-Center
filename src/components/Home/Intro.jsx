// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import Skeleton from 'react-loading-skeleton'
import IntroInlineEditor from '../editor/IntroInlineEditor'
import { fetchInfoPage } from 'actions/page'
import { Flex, Box } from 'rebass'

type Props = {
  auth: Object,
  fetchInfoPage: Function,
  page: Object,
  isFetching: boolean
}

class Intro extends Component<Props> {
  displayName = 'homepage introduction'
  // set defaultprops to prevent console warnings
  static defaultProps = {
    page: {
      data: {
        attributes: {}
      }
    }
  }
  componentDidMount() {
    this.props.fetchInfoPage('dsc-intro')
  }
  render() {
    const { isFetching, page } = this.props

    if (!isFetching && page.data.attributes.content) {
      return <IntroInlineEditor auth={this.props.auth} page={this.props.page} />
    }
    return (
      <Flex justify="center">
        <Box w={'95%'}>
          <Skeleton count={5} />
        </Box>
      </Flex>
    )
  }
}

const mapStateToProps = state => {
  const slugName = 'dsc-intro'
  return {
    auth: state.auth,
    isFetching: state.page.isFetching,
    page: state.page[slugName]
  }
}

export default connect(mapStateToProps, { fetchInfoPage })(Intro)
