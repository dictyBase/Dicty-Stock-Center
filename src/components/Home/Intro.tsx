import React from "react"
import { useQuery } from "@apollo/react-hooks"
import Grid from "@material-ui/core/Grid"
import Skeleton, { SkeletonTheme } from "react-loading-skeleton"
import InlineEditor from "components/InlineEditor"
import { GET_CONTENT_BY_SLUG } from "queries/queries"

const slugName = "dsc-intro"

/**
 * Intro fetches and displays the Intro page content.
 */

const Intro = () => {
  const { loading, error, data } = useQuery(GET_CONTENT_BY_SLUG, {
    variables: {
      slug: slugName,
    },
  })

  if (loading) {
    return (
      <Grid container justify="center">
        <Grid item xs={12}>
          <SkeletonTheme color="#D3D3D3	" highlightColor="#DCDCDC">
            <Skeleton count={5} />
            <br />
            <br />
          </SkeletonTheme>
        </Grid>
      </Grid>
    )
  }

  if (error) {
    return <div>got an error</div>
  }

  return <InlineEditor data={data.contentBySlug} />
}

export default Intro
