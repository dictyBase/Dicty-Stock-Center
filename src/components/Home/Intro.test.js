import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Intro from './Intro'

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Intro />, div)
})

test('matching a snapshot of Intro', () => {
    const component = renderer.create(<Intro />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
