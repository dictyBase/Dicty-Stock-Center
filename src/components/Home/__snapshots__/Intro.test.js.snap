// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Intro 1`] = `
<div
  className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="sc-bdVaJa eJiGdf"
    is={null}
  >
    <div
      className="react-loading-theme-0"
    >
      <span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
      </span>
      <br />
      <br />
    </div>
  </div>
</div>
`;
